Sources/ChessKit/Position.swift

```diff
@@ -198,7 +208,20 @@ public struct Position: Equatable, Sendable {
             return false
         }
     }
-
+    
+    /// Checks if a position already occurred N times
+    ///
+    ///   - parameter times: the times the position occurred
+    ///   - parameter positions: all the positions occurred during the game
+    /// - Returns: a bool stating if the position already occurred N times
+    public func occurred(times: Int, in positions: [Int:Int]) -> Bool {
+        if positions[hashed] == times {
+            return true
+        }
+        
+        return false
+    }
```

Comment:
If `Position` is `Hashable` this can be removed and you can do everything from inside `Board`